### PR TITLE
Support more `Number` types

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

**Review by saket — APPROVED** [review]
Good idea!

**Comment by Goooler** [comment]
Clean up a bit.

File: library/src/main/kotlin/me/saket/filesize/FileSize.kt [hunk 1]
```diff
@@ -76,38 +84,61 @@ class FileSize(val bytes: Long): Comparable<FileSize> {
...
-      get() = FileSize(bytes = (this * BytesPerGb))
+    inline val Number.bytes: FileSize
+      get() {
+        val result = when (this) {
+          is Byte -> this.toLong()
+          is Short -> this.toLong()
+          is Int -> this.toLong()
+          is Long -> this
+          is Float -> this.toLong()
+          is Double -> this.toLong()
```
Comment (saket) [inline library/src/main/kotlin/me/saket/filesize/FileSize.kt]: `Double#bytes` currently produces a compilation error:

https://github.com/saket/file-size/blob/893a7c62a75dce28b7c8e446d78e916923d680bf/library/src/main/kotlin/me/saket/filesize/FileSize.kt#L112-L119

how can `Number#bytes` handle this?

Comment (Goooler) [reply library/src/main/kotlin/me/saket/filesize/FileSize.kt]: It's explicit.

image

Comment (saket) [reply library/src/main/kotlin/me/saket/filesize/FileSize.kt]: LGTM!

**Comment by saket** [comment]
Thanks @Goooler!

**Pull request merged**